developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,300 @@
+
+In Typescript, this is how a single signer authenticator looks like:
+```typescript
+interface Authenticator {
+  public_key: Uint8Array,
+  signature: Uint8Array
+}
+```
+
+A multisig authenticator example is shown below:
+```typescript
+interface  MultiEd25519PublicKey {
+  public_keys: Uint8Array[], // A list of public keys
+  threshold: Uint8, // At least `threshold` signatures must be valid
+}
+
+interface MultiEd25519Signature {
+    signatures: Uint8Array[], // A list of signatures
+    // 4 bytes, at most 32 signatures are supported.
+    // If Nth bit value is `1`, the Nth signature should be provided in `signatures`. Bits are read from left to right
+    bitmap: Uint8Array,
+}
+
+interface MultisigAuthenticator {
+  public_key: MultiEd25519PublicKey,
+  signature: MultiEd25519Signature
+}
+```
+
+## Detailed steps
+
+1. Create a RawTransaction.
+2. Create a signing message.
+3. Sign the signing message with account private key.
+4. Create a SignedTransaction.
+5. Serialize SignedTransaction into bytes with BCS, and then hex-encode the bytes into a string.
+
+### Step 1. Create a RawTransaction
+
+The below example assumes the transaction has a script function payload.
+
+```typescript
+
+interface ModuleId {
+  address: string,
+  name: string
+}
+
+interface ScriptFunction {
+  module: ModuleId,
+  function: string,
+  ty_args: string[],
+  args: Uint8Array
+}
+
+interface RawTransaction {
+  sender: string,
+  sequence_number: number,
+  payload: ScriptFunction,
+  max_gas_amount: number,
+  gas_unit_price: number,
+  expiration_timestamp_secs: number,
+  chain_id: number,
+}
+
+function createRawTransaction(): RawTransaction {
+  const payload: ScriptFunction = {
+    module: {
+      address: "00000000000000000000000000000001",
+      name: "TestCoin"
+    },
+    function: "transfer",
+    ty_args: [],
+    args: [
+      BSC.serialize_str("00000000000000000000000000000002"), // receipient of the transfer
+      BSC.serialize_uint64(2), // amount to transfer
+    ]
+  }
+
+  return {
+    "sender": "00000000000000000000000000000001",
+    "sequence_number": 1,
+    "max_gas_amount": 2000,
+    "gas_unit_price": 1,
+    // Unix timestamp, in seconds + 10 minutes
+    "expiration_timestamp_secs": Math.floor(Date.now() / 1000) + 600,
+    "payload": payload,
+    "chain_id": 3
+  };
+}
+
+```
+
+### Step 2. Create a signing message
+
+1. [SHA3_256][sha3] hash bytes of string `APTOS::RawTransaction`.
+2. Bytes of BCS serialized RawTransaction.
+3. Concat the prefix and BCS bytes.
+
+```typescript
+
+function hashPrefix(): Buffer {
+  let hash = SHA3.sha3_256.create();
+  hash.update(`APTOS::RawTransaction`);
+  return Buffer.from(hash.arrayBuffer());
+}
+
+function bcsSerializeRawTransaction(txn: RawTransaction): Buffer {
+  ...
+}
+
+// This will hash a raw transaction into bytes
```

Comment:
this isn't right, you're just hashing a transaction prefix and appending the raw transaction